app/services/liquid_assigns_service.rb

```diff
@@ -1,12 +1,15 @@
 class LiquidAssignsService
-  attr_accessor :project
+  AVAILABLE_PROJECT_ASSIGNS = %w{ evidences issues nodes notes tags }.freeze
 
-  def initialize(project)
+  attr_accessor :project, :text
+
+  def initialize(project:, text: nil)
     @project = project
```

Comment:
What about passing the `project_id` instead and finding it here?

Reply:
I'm not sure if we're winning with this change if we have to update every instance of this initialization as well as possibly move the `can?` check. See `liquid_enabled_resource`:
```
  def project_assigns
    # This is required because we may be in Markup#preview that's passing
    # :project_id for Tylium rendered editors
    project = Project.find(params[:project_id])
    authorize! :use, project

    LiquidAssignsService.new(project: project, text: params[:text]).assigns
```

The check requires a reference to the `current_user`